contributors/design-proposals/local-storage-pv.md

```diff
@@ -0,0 +1,631 @@
+settings will be applied on the path.  However, multiple pods
+using the same volume should use the same fsGroup.
+* Volume lifecycle is not clearly defined, and the volume has to be manually
+cleaned up by users.  For Local volumes, the PV has a clearly defined
+lifecycle.  Upon PVC deletion, the PV will be released (if it has the Delete
+policy), and all the contents under the path will be deleted.  In the future,
+advanced cleanup options, like zeroing can also be specified for a more
+comprehensive cleanup.
+
+##### API Changes
+
+All new changes are protected by a new feature gate, `PersistentLocalVolumes`.
+
+A new `LocalVolumeSource` type is added as a `PersistentVolumeSource`.  For this
+initial phase, the path can only be a mount point or a directory in a shared
+filesystem.
+
+```
+type LocalVolumeSource struct {
+        // The full path to the volume on the node
+        // For alpha, this path must be a directory
+        // Once block as a source is supported, then this path can point to a block device
+        Path string
+}
+
+type PersistentVolumeSource struct {
+    <snip>
+    // Local represents directly-attached storage with node affinity.
+    // +optional
+    Local *LocalVolumeSource
+}
+```
+
+The relationship between a Local volume and its node will be expressed using
+PersistentVolume node affinity, described in the following section.
+
+Users request Local volumes using PersistentVolumeClaims in the same manner as any
+other volume type. The PVC will bind to a matching PV with the appropriate capacity,
+AccessMode, and StorageClassName.  Then the user specifies that PVC in their
+Pod spec.  There are no special annotations or fields that need to be set in the Pod
+or PVC to distinguish between local and remote storage.  It is abstracted by the
+StorageClass.
+
+#### PersistentVolume Node Affinity
+
+PersistentVolume node affinity is a new concept and is similar to Pod node affinity,
+except instead of specifying which nodes a Pod has to be scheduled to, it specifies which nodes
+a PersistentVolume can be attached and mounted to, influencing scheduling of Pods that
+use local volumes.
+
+For a Pod that uses a PV with node affinity, a new scheduler predicate
+will evaluate that node affinity against the node's labels.  For this initial phase, the
+PV node affinity is only considered by the scheduler for already-bound PVs.  It is not
+considered during the initial PVC/PV binding, which will be addressed in a future release.
+
+Only the `requiredDuringSchedulingIgnoredDuringExecution` field will be supported.
+
+##### API Changes
+
+For the initial alpha phase, node affinity is expressed as an optional
+annotation in the PersistentVolume object.
+
+```
+// AlphaStorageNodeAffinityAnnotation defines node affinity policies for a PersistentVolume.
+// Value is a string of the json representation of type NodeAffinity
+AlphaStorageNodeAffinityAnnotation = "volume.alpha.kubernetes.io/node-affinity"
```

Comment:
is the plan to migrate the annotation to a field on the PV spec or drop the alpha annotation when the field is added?